test/clj_kondo/main_test.clj

```diff
@@ -2374,26 +2373,11 @@ i/blank? (i/starts-with?) i/x
                        :type-mismatch {:level :error}}})))
   (is (empty? (lint! "
 (ns dev.clj-kondo {:clj-kondo/config '{:linters {:missing-docstring {:level :warning}}}}
-(:require [potemkin :refer [import-vars]]))
+(:require [potemkin :refer [import-vars]]
+          [clojure.string]))
 
 (import-vars [clojure.string blank?, starts-with?, ends-with?, includes?])"
                      {:linters {:unresolved-symbol {:level :error}}})))
-  (is (empty? (lint! "
-(ns foo.bar)
-
-(defn foo []) ;; non-empty to generated ns cache
-
-;; dynamically generated baz
```

Comment:
@borkdude I removed this test because AFAICS this code doesn't compile, I didn't see any dynamically generated baz that would result in a valid `foo.bar/baz`, am I missing something?

Reply:
The dynamically generated baz is a hypothetical one: suppose there was a macro in this namespace that generated it. Then the below code should still work, although clj-kondo did not see the baz var.

Reply:
Via git blame you can see that this test was added for https://github.com/clj-kondo/clj-kondo/issues/1167.

Reply:
I see, this is not possible anymore with the current code right? unless we ad the macro that would expand to a var-definition, what do you think?

Reply:
> I see, this is not possible anymore with the current code right? 

Explain?

Reply:
my bad, it's possible, it is just that we would need to include the macro to the test keep working, do you think that would be enough?

Reply:
I think you would know better than me since you reported #1167 :)

Reply:
Feel free to change the test, ensuring that it still fixes that issues.

Reply:
makes sense! will do

Reply:
Done @borkdude

Reply:
@ericdallo That is still not a correct fix. The point was that there could be a dynamic creation of the `baz` var in the namespace. You are introducing a static creation of the `baz` var.

Reply:
Hum, so I'm not sure I understood how this dynamic creation works

Reply:
The issue is that we get a foo.bar/baz unresolved with my changes, that it seems correct given this code, doesn't ?

Reply:
The fix for #1167 was that we should just assume the imported var to be there, even if it didn't exist. I'm fine with dropping this behavior, but your reported it yourself, so it may backfire on you.